### ISSUE TITLE
Implement ListDataset to iterate over files

### ISSUE THREAD

**Issue by DaloroAT** [OML-Team/open-metric-learning#205]
This issue is the part of #85 

## Design

`ListDataset` should be very simple. It can iterate over files, apply transforms, and return tensors.

``` python
from torch.utils.data import Dataset
import torch

class ListDataset(Dataset):
    def __init__(self, filenames_list, transforms, f_imread):
        ...

    def __getitem__(self, idx) -> torch.Tensor:
        ...
```

Please, don't forget to support different types of transforms. Now we use transforms from `albumentations` and default `torchvision` transforms, you can check it in `oml.datasets.base.BaseDataset`.

Transforms are responsible for augmentations of the original images, resizing and normalizing.

## Test

Iterate over the files in mock dataset with dataloader and check shapes of batches. 

Mock dataset can be found in 
``` python
from oml.const import MOCK_DATASET_PATH
print(MOCK_DATASET_PATH / 'images')
```
If you have no mock dataset locally, use `make download_mock_dataset` from the root of the project.

**Comment by DaloroAT** [comment]
@churnikov @AlekseySh

**Comment by churnikov** [comment]
Hi!

I would like to check the following, is this script going to only work with crops of images, or I should load bounding boxes from some place?

**Comment by DaloroAT** [comment]
Perhaps we can support the use of bounding boxes from JSON, CSV, or whatever.

Right now you can add one more argument `bounding_boxes = [[x1,y1,x2,y2], ...]` into `__init__`.

**Comment by churnikov** [comment]
https://github.com/OML-Team/open-metric-learning/pull/210

**Comment by churnikov** [comment]
@AlekseySh 

Thank you for the approval and merge!

I saw on email that I forgot to add ListDataset to documentation. I fixed it in this pull request
https://github.com/OML-Team/open-metric-learning/pull/224
